tls/s2n_security_policies.c

```diff
@@ -891,6 +900,30 @@ int s2n_config_set_cipher_preferences(struct s2n_config *config, const char *ver
     return 0;
 }
 
+static bool s2n_cipher_suite_uses_chacha20_alg(struct s2n_cipher_suite *cipher_suite) {
+    if (!cipher_suite) {
+        return false;
+    }
+
+    /* If s2n_chacha20_poly1305 is available, then we can directly check the record_alg */
+    if (s2n_chacha20_poly1305.is_available()) {
+        return cipher_suite->record_alg && cipher_suite->record_alg->cipher == &s2n_chacha20_poly1305;
+    } 
```

Comment:
If you're going to do this, you should call s2n_cipher_suite_uses_chacha20_alg instead of recreating its logic.